zone_cache_settings.go

```diff
@@ -0,0 +1,71 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+// ZoneSetting contains settings for a zone.
+type ZoneCacheSetting struct {
+	ID         string      `json:"id"`
+	ModifiedOn string      `json:"modified_on,omitempty"`
+	Value      interface{} `json:"value"`
```

Comment:
this type is too loose i'm afraid. we'll need to match the request/response body a bit closer - https://api.cloudflare.com/#zone-cache-settings-change-variants-setting

we should be able to add each of the image formats to a `[]string`

Reply:
@jacobbednarz I could do that, and make the new functions specific to cache variants (i.e. `ZoneCacheVariants` instead of `ZoneCacheSettings`), but it would not be consistent with the approach taken with `ZoneSingleSetting` / `UpdateZoneSingleSetting`. I think the upside is that if new cache settings are added in the future, they would get automatically supported. The downside is less clarity around what's expected to be passed as the `Value` here / less type validation.

The cache setting endpoints are documented about the same as the zone setting endpoints, so I thought it'd make sense to use the same approach

Reply:
To be clear, I don't have a strong preference. I'm just trying to get clarity around if you'd like to really expose cache variants as a first-class resource, or consider it as a sub-type of a "Zone Cache Setting".

I guess this would also influence how we should expose this through the terraform provider (i.e. should we have an heavily typed `cloudflare_zone_cache_variants` resource, or rather a `cloudflare_zone_cache_settings`, with a `variants` property embedded in it?)

Reply:
please please please don't use zone settings a good example to work from. it suffers from a few problems (loose types, old endpoints) which make maintenance really difficult and hard to extend. if you're after a good recent, example check out `rulesets.go`.

Reply:
all good, thanks for the clarifications 😄 

I'll add some typings and make the functions more specific to cache variants.

You suggested the following struct:

```
type ZoneCacheSettingValues struct {
	Avif []string `json:"avif"`
	Bmp  []string `json:"bmp"`
	Gif  []string `json:"gif"`
	Jpeg []string `json:"jpeg"`
	Jpg  []string `json:"jpg"`
	Jpg2 []string `json:"jpg2"`
	Jp2  []string `json:"jp2"`
	Png  []string `json:"png"`
	Tiff []string `json:"tiff"`
	Tif  []string `json:"tif"`
	Webp []string `json:"webp"`
}

type ZoneCacheSetting struct {
	ID         string                  `json:"id"`
	ModifiedOn time.Time               `json:"modified_on"`
	Value      ZoneCacheSettingValues  `json:"value"`
}
```

A downside of this approach is that if a file extension is added, then it would also need to be added to both the terraform provider and cloudflare-go. An alternative would be to just use `map[string][]string`. WDYT?

Reply:
yep, that is intentional behaviour. just because a field/value comes back in the API doesn't mean it is yet ready for public consumption. Cloudflare is API driven and until the service teams 👍 the field for public use, we don't want end users depending on it.

Reply:
All good! I pushed a few commits to address the feedback you provided. Ready for a re-review!